edit-post/components/block-settings-menu/plugin-block-settings-menu-group.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createSlotFill } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+const { Fill: PluginBlockSettingsMenuGroup, Slot } = createSlotFill( 'PluginBlockSettingsMenuGroup' );
+
+const PluginBlockSettingsMenuGroupSlot = ( { fillProps, selectedBlocks } ) => {
+	selectedBlocks = map( selectedBlocks, ( block ) => block.name );
+	return (
+		<Slot fillProps={ { ...fillProps, selectedBlocks } } >
+			{ ( fills ) => ! isEmpty( fills ) && (
+				<Fragment>
+					<div className="editor-block-settings-menu__separator" />
+					{ fills }
+				</Fragment>
+			) }
+		</Slot>
+	);
+};
+
+PluginBlockSettingsMenuGroup.Slot = withSelect( ( select, { fillProps: { clientIds } } ) => ( {
+	selectedBlocks: select( 'core/editor' ).getBlocksByUID( clientIds ),
```

Comment:
`getBlocksByUID` was deprecated in Gutenberg v3.3 and is slated to be removed in the next release. A warning would have been logged to your console on its use.